### ISSUE TITLE
Training loss higher than testing loss?

### ISSUE THREAD

**Issue by samin-batra** [mrdbourke/pytorch-deep-learning#689]
Hello,

I was going through the exercise for the 2nd module (linear regression). I have noticed that when I specify the weights and bias parameters as nn.Parameters, my training loss is higher than my testing loss - but when I declare the layers using the nn.Layer object, the issue does not occur. Consequently, the parameters are not converging well. Has anyone else faced the same issue? 
Here's the code:
1) By defining the parameters using nn.Parameter
class LinearModelEx(nn.Module):
  def __init__(self):
    super().__init__()
    self.weights = nn.Parameter(torch.randn(1,requires_grad=True,dtype=torch.float32))
    self.bias = nn.Parameter(torch.randn(1,requires_grad=True,dtype=torch.float32))
   
  def forward(self,x:torch.Tensor):
    return self.weights*x+self.bias
    
2) By defining the parameters using nn.Layer:
class LinearModelEx(nn.Module):
  def __init__(self):
    super().__init__()
    self.linear_layer = nn.Linear(in_features=1,out_features=1)
  
  def forward(self,x:torch.Tensor):
    return self.linear_layer(x)

The code for the training loop is the same.
